### PR TITLE
Add Kegen G-Boy layout support

### PR DESCRIPTION
**Pull request by mattchan**
## Description

This adds support for the Kegen G-Boy PCB which includes the following
* Split backspace
* Full right shift
* ISO enter
* Split left shift
* Various bottom row layouts
* QMK lighting controls

## QMK Pull Request 

https://github.com/qmk/qmk_firmware/pull/18048

## Checklist

- [x] The VIA support for this keyboard is in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

### CODE REVIEW

**Comment by mattchan** [comment]
I'll add v3 in a separate commit and pull request once this is approved.

File: src/kegen/gboy/gboy.json [hunk 1]
```diff
@@ -0,0 +1,309 @@
+{
+    "name": "G-Boy",
+    "vendorId": "0x4b65",
+    "productId": "0x6762",
+    "firmwareVersion": 0,
+    "lighting": "qmk_backlight_rgblight",
+    "keycodes": [
```
Comment (yiancar) [inline src/kegen/gboy/gboy.json:7]: Can you remove this, its not needed for v2

File: src/kegen/gboy/gboy.json [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    "lighting": "qmk_backlight_rgblight",
+    "keycodes": [
+        "via/keycodes",
+        "via/qmk_lighting"
+    ],
+    "matrix": {
+        "rows": 5,
+        "cols": 16
+    },
+    "menus": [
```
Comment (yiancar) [inline src/kegen/gboy/gboy.json:15]: can you remove this, its not needed for v2

File: src/kegen/gboy/gboy.json (hunk 1 above)
Comment (yiancar) [reply src/kegen/gboy/gboy.json:7]: also same for "firmwareVersion": 0,

**Comment by mattchan** [comment]
I've updated the file with the suggested changes.

**Comment by Xelus22** [comment]
LGTM

**Pull request merged**